### PR TITLE
Fix deeplink redirect to login

### PR DESCRIPTION
**Pull request by bogdan-sava**
Unauthenticated deep link should redirect to login with redirect query parameter to get back after successful authentication.

### CODE REVIEW

File: new/egeria-app.container.js [hunk 1]
```diff
@@ -33,34 +33,13 @@ class EgeriaApp extends PolymerElement {
...
-      const hasRoles = roles !== null &&  roles.status === 200;
-      const hasAppInfo = appInfo !== null && appInfo.status === 200;
-
-      if(![hasComponents, hasCurrentUser, hasRoles, hasAppInfo].includes(false)) {
-        this.isLoading = true;
-      } else {
-        this.isLoading = false;
-      }
-    });
+    if( !['login', 'forbidden', 'error', 'empty'].includes(path) )
```
Comment (sarbull) [inline new/egeria-app.container.js:39]: can you add brackets? i'll embrace that

File: new/egeria-app.container.js [hunk 2]
```diff
@@ -33,34 +33,13 @@ class EgeriaApp extends PolymerElement {
   constructor() {
     super();
 
-    this.isLoading = true;
+    this.isLoading = false;
+    let path = window.location.pathname.substr(1)
```
Comment (sarbull) [inline new/egeria-app.container.js:37]: can you also add ; at the end, it will make a point

File: new/egeria-app.container.js [hunk 3]
```diff
@@ -89,6 +68,37 @@ class EgeriaApp extends PolymerElement {
     ];
   }
 
+  /**
+   * performes requests needed by authenticated page
+   * @private
+   */
```
Comment (sarbull) [inline new/egeria-app.container.js:74]: 👍

File: new/egeria-app.container.js [hunk 4]
```diff
@@ -89,6 +68,37 @@ class EgeriaApp extends PolymerElement {
...
+      this.isLoggedIn = currentUser !== null && currentUser.status === 200;
+
+      const hasComponents = components !== null && components.status === 200;
+      const hasCurrentUser = currentUser !== null && currentUser.status === 200;
+      const hasRoles = roles !== null &&  roles.status === 200;
+      const hasAppInfo = appInfo !== null && appInfo.status === 200;
+
+      if(![hasComponents, hasCurrentUser, hasRoles, hasAppInfo].includes(false)) {
+        this.isLoading = true;
+      }
```
Comment (sarbull) [inline new/egeria-app.container.js:98]: an else is not needed here as it was previously?

File: new/egeria-app.container.js (hunk 1 above)
Comment (bogdan-sava) [reply new/egeria-app.container.js]: done :)

File: new/egeria-app.container.js (hunk 2 above)
Comment (bogdan-sava) [reply new/egeria-app.container.js]: done

File: new/egeria-app.container.js (hunk 4 above)
Comment (bogdan-sava) [reply new/egeria-app.container.js:99]: not needed as it was initialised with false. 
But as long I extracted as a separate method, I should use it, or even better assign it directly. 
No need to have an if  statement only to assign true or false. :)

**Review by sarbull — APPROVED** [review]

**Pull request merged**